### DISCUSSION TITLE
Use tables without using Eloquent

### DISCUSSION THREAD

**Discussion by yablochnyi** [filamentphp/filament#11078] [Help] [answered]
### Package

Table builder

### Package Version

v3

### How can we help you?

Is it possible to use the table component without Eloquent? I'm using a sushi package, but I need to fetch the data based on the ID, how can I transfer it?
```
public function getRows()
    {
        $userDataMarketPositions = Cache::get("data:market_positions:" . $id);
        return $userDataMarketPositions['data'];
    }
```

**Comment by phh** [comment]
I think it's a good idea and seems like it's something that is already on the 2024 roadmap :)

https://filamentphp.com/community/alexandersix-filament-what-to-expect-in-2024#static-table-data-sources

**Comment by cntabana** [comment, marked as answer]
Hello, here's how you pass parameters in Sushi:


```
 protected static $id;
 public static function setVar($id)
 {
        self::$id = $id;

        return self::query();
  }


public function getRows()
 {
        $userDataMarketPositions = Cache::get("data:market_positions:" . self::$id);
        return $userDataMarketPositions['data'];
 }
```



Then, on the Filament page, you can proceed as follows:

```
protected function getTableQuery(): Builder
 {
        YourSushiModel::setVar($id);
        return YourSushiModel::query();
 }
```

Hopefully, this is what you are looking for.

**Reply by yablochnyi** [reply]
thanks
